lib/llvmopencl/Barrier.h

```diff
@@ -60,8 +62,11 @@ namespace pocl {
 
       llvm::Function *F = llvm::cast<llvm::Function>
         (M->getOrInsertFunction(BARRIER_FUNCTION_NAME,
-                                llvm::Type::getVoidTy(M->getContext()),
-                                NULL));
+                                llvm::Type::getVoidTy(M->getContext())
+#if LLVM_OLDER_THAN_5_0
```

Comment:
It might be cleaner (in terms of readability) to duplicate the whole function call instead of guarding the single parameter.